src/load-modules.ts

```diff
@@ -76,81 +94,133 @@ const nameFormatters: Record<string, NameFormatter> = {
  * @param  {(string, ModuleDescriptor) => string} opts.formatName
  * Used to format the name the module is registered with in the container.
  *
+ * @param  {boolean} opts.esModules
+ * Set to `true` to use Node's native ECMAScriptModules modules
+ *
  * @return {Object}
  * Returns an object describing the result.
  */
-export function loadModules(
+export function loadModules<ESM extends boolean>(
   dependencies: LoadModulesDeps,
   globPatterns: string | Array<string | GlobWithOptions>,
-  opts?: LoadModulesOptions
-) {
+  opts?: LoadModulesOptions<ESM>
+): Promise<LoadModulesResult> | LoadModulesResult {
   const container = dependencies.container
   opts = optsWithDefaults(opts, container)
   const modules = dependencies.listModules(globPatterns, opts)
 
-  const result = modules.map((m) => {
-    const items: Array<{
-      name: string
-      path: string
-      opts: object
-      value: unknown
-    }> = []
+  if (opts?.esModules) {
+    return loadEsModules(dependencies, container, modules, opts)
+  } else {
+    const result = modules.map((m) => {
+      const loaded = dependencies.require(m.path)
+      return parseLoadedModule(loaded, m)
+    })
+    return registerModules(result, container, modules, opts)
+  }
+}
 
-    const loaded = dependencies.require(m.path)
+/**
+ * Loads the modules using native ES6 modules and the async import()
+ * @param {AwilixContainer} container
+ * @param {ModuleDescriptor[]} modules
+ * @param {LoadModulesOptions} opts
+ */
+async function loadEsModules<ESM extends boolean>(
+  dependencies: LoadModulesDeps,
+  container: AwilixContainer,
+  modules: ModuleDescriptor[],
+  opts: LoadModulesOptions<ESM>
+): Promise<LoadModulesResult> {
+  const importPromises = []
+  for (const m of modules) {
+    importPromises.push(dependencies.require(m.path))
```

Comment:
Hi @jeffijoe and @richardsimko 
Should i open this comment as a new issue, did you get what i mean?

I used the following glob that throws that error in windows
![image](https://user-images.githubusercontent.com/3621676/104539384-27a24e80-55fc-11eb-8965-fcd78033cb82.png)

Reply:
Hi! I thought node would handle that automatically! It does on macOS at least, since the path that `import()` gets called with (Which is at the top of the stack from this call) is a `file://` URL. I sadly don't have access to a Windows computer to test but reading this https://github.com/nodejs/node/issues/31710 the recommended approach seems to be to use `URL.pathToFileURL` https://nodejs.org/api/url.html#url_url_pathtofileurl_path.

Reply:
@richardsimko, how could i use it in loadModules method?

Reply:
I don't think you can do anything as a consumer, but you can submit a PR here to fix it :)